apollo/integrations/databricks/databricks_sql_warehouse_proxy_client.py

```diff
@@ -20,8 +25,28 @@ def __init__(self, credentials: Optional[Dict], **kwargs: Dict):
             raise ValueError(
                 f"Databricks agent client requires {_ATTR_CONNECT_ARGS} in credentials"
             )
+
+        if self._credentials_use_oauth(credentials[_ATTR_CONNECT_ARGS]):
+            credentials[_ATTR_CONNECT_ARGS][_ATTR_CREDENTIALS_PROVIDER] = (
+                self._oauth_credentials_provider(credentials[_ATTR_CONNECT_ARGS])
+            )
+
         self._connection = sql.connect(**credentials[_ATTR_CONNECT_ARGS])
 
+    def _credentials_use_oauth(self, connect_args: Dict) -> bool:
+        return CLIENT_ID_KEY in connect_args and CLIENT_SECRET_KEY in connect_args
+
+    def _oauth_credentials_provider(self, connect_args: Dict) -> Callable:
+        # create the auth callable here because it can't be serialized
+        config = Config(
+            host=connect_args.get("server_hostname"),
```

Comment:
nit: super minor and just for consistency I'd define a constant for `server_hostname`